### PR TITLE
Skip prereleases and postreleases in version check

### PR DESCRIPTION
**Pull request by davisagli**
# Critical Changes

# Changes
* Skip beta releases when checking for a newer cumulusci version

# Issues Closed

### CODE REVIEW

File: cumulusci/cli/cci.py [hunk 1]
```diff
@@ -65,13 +65,16 @@ def timestamp_file():
         yield f
 
 
-def get_latest_version():
+def get_latest_final_version():
```
Comment (cdcarter) [inline cumulusci/cli/cci.py]: is "final" a PyPI term? nothing about this is...final...per se...

File: cumulusci/cli/cci.py [hunk 2]
```diff
@@ -65,13 +65,16 @@ def timestamp_file():
...
     """ return the latest version of cumulusci in pypi, be defensive """
```
Comment (cdcarter) [inline cumulusci/cli/cci.py]: during the refactor the "be defensive" got moved into the calling method. update docstring plz

File: cumulusci/cli/cci.py [hunk 3]
```diff
@@ -65,13 +65,16 @@ def timestamp_file():
...
     # use the pypi json api https://wiki.python.org/moin/PyPIJSON
     res = requests.get("https://pypi.org/pypi/cumulusci/json", timeout=5).json()
     with timestamp_file() as f:
         f.write(bytes(time.time()))
-    return pkg_resources.parse_version(res["info"]["version"])
+    versions = [pkg_resources.parse_version(v) for v in res["releases"].keys()]
```
Comment (cdcarter) [inline cumulusci/cli/cci.py]: we already have a LOT of cci releases and more coming, is there any benefit to only parsing and checking the last N? or is this all so fast we dn't care

File: cumulusci/cli/cci.py (hunk 1 above)
Comment (davisagli) [reply cumulusci/cli/cci.py]: It's from PEP 440: https://www.python.org/dev/peps/pep-0440/#final-releases

File: cumulusci/cli/cci.py (hunk 3 above)
Comment (davisagli) [reply cumulusci/cli/cci.py]: Since they seem to be sorted alphabetically and we can have many betas, I'm not sure that's safe.

**Pull request merged**